packages/repository/test/unit/repository-mixin/repository-mixin.test.ts

```diff
@@ -0,0 +1,125 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {
+  RepositoryMixin,
+  juggler,
+  DataSourceConstructor,
+} from '../../../';
+import {Application} from '@loopback/core';
+
+// tslint:disable:no-any
+
+describe('RepositoryMixin', () => {
+  const appWithRepoMixin = RepositoryMixin(Application);
+
+  class NoteRepo {
+    model: any;
+
+    constructor() {
+      const ds: juggler.DataSource = new DataSourceConstructor({
+        name: 'db',
+        connector: 'memory',
+      });
+
+      this.model = ds.createModel(
+        'note',
+        {title: 'string', content: 'string'},
+        {},
+      );
+    }
+  }
+
+  it('mixed class has .repository()', () => {
+    const myApp = new appWithRepoMixin();
+    expect(typeof myApp.repository).to.be.eql('function');
+  });
+
+  it('binds repository from constructor', () => {
+    const myApp = new appWithRepoMixin({
+      repositories: [NoteRepo],
+    });
+
+    const boundRepositories = myApp.find('repositories.*').map(b => b.key);
+    expect(boundRepositories).to.containEql('repositories.NoteRepo');
+
+    const repoInstance = myApp.getSync('repositories.NoteRepo');
+    expect(repoInstance).to.be.instanceOf(NoteRepo);
+  });
+
+  it('binds repository from app.repository()', () => {
+    const myApp = new appWithRepoMixin();
+
+    const boundReposBefore = myApp.find('repositories.*').map(b => b.key);
+    expect(boundReposBefore).to.be.empty();
+
+    myApp.repository(NoteRepo);
+
+    const boundReposAfter = myApp.find('repositories.*').map(b => b.key);
+    expect(boundReposAfter).to.containEql('repositories.NoteRepo');
+
+    const repoInstance = myApp.getSync('repositories.NoteRepo');
+    expect(repoInstance).to.be.instanceOf(NoteRepo);
+  });
+
+  it('binds user defined component without repository', () => {
+    class TestComponent {}
+
+    const myApp = new appWithRepoMixin({
+      components: [TestComponent],
+    });
+
+    const componentKeys = myApp.find('components.*').map(b => b.key);
+    expect(componentKeys).to.containEql('components.TestComponent');
+    const componentInstance = myApp.getSync('components.TestComponent');
+    expect(componentInstance).to.be.instanceOf(TestComponent);
+  });
+
+  it('binds user defined component with repository in constructor', () => {
+    class TestComponent {
+      repositories = [NoteRepo];
+    }
+
+    const myApp = new appWithRepoMixin({
+      components: [TestComponent],
+    });
+
+    const componentKeys = myApp.find('components.*').map(b => b.key);
+    expect(componentKeys).to.containEql('components.TestComponent');
+    const componentInstance = myApp.getSync('components.TestComponent');
+    expect(componentInstance).to.be.instanceOf(TestComponent);
```

Comment:
I think these four checks are redundant (the previous test has already verified that a component is correctly loaded), and should be removed.